crates/bevy_render/src/render_resource/bind_group_layout_entries.rs

```diff
@@ -0,0 +1,582 @@
+///             // Screen texture
+///             texture_2d(TextureSampleType::Float { filterable: true }),
+///             // Sampler
+///             sampler(SamplerBindingType::Filtering),
+///         ),
+///     ),
+/// );
+/// ```
+///
+/// instead of
+///
+/// ```ignore
+/// let layout = render_device.create_bind_group_layout(
+///     "my_bind_group_layout",
+///     &[
+///         // Screen texture
+///         BindGroupLayoutEntry {
+///             binding: 0,
+///             visibility: ShaderStages::FRAGMENT,
+///             ty: BindingType::Texture {
+///                 sample_type: TextureSampleType::Float { filterable: true },
+///                 view_dimension: TextureViewDimension::D2,
+///                 multisampled: false,
+///             },
+///             count: None,
+///         },
+///         // Sampler
+///         BindGroupLayoutEntry {
+///             binding: 1,
+///             visibility: ShaderStages::FRAGMENT,
+///             ty: BindingType::Sampler(SamplerBindingType::Filtering),
+///             count: None,
+///         },
+///     ],
+/// );
+/// ```
+///
+/// or
+///
+/// ```ignore
+/// render_device.create_bind_group_layout(
+///     "my_bind_group_layout",
+///     &BindGroupLayoutEntries::single(
+///         ShaderStages::FRAGMENT,
+///         texture_2d(TextureSampleType::Float { filterable: true }),
+///     ),
+/// );
+/// ```
+///
+/// instead of
+///
+/// ```ignore
+/// let layout = render_device.create_bind_group_layout(
+///     "my_bind_group_layout",
+///     &[
+///         BindGroupLayoutEntry {
+///             binding: 0,
+///             visibility: ShaderStages::FRAGMENT,
+///             ty: BindingType::Texture {
+///                 sample_type: TextureSampleType::Float { filterable: true },
+///                 view_dimension: TextureViewDimension::D2,
+///                 multisampled: false,
+///             },
+///             count: None,
+///         },
+///     ],
+/// );
+/// ```
+
+#[derive(Clone, Copy)]
+pub struct BindGroupLayoutEntryBuilder {
+    ty: BindingType,
+    visibility: Option<ShaderStages>,
+    count: Option<NonZeroU32>,
+}
+
+impl BindGroupLayoutEntryBuilder {
+    pub fn visibility(mut self, visibility: ShaderStages) -> Self {
+        self.visibility = Some(visibility);
+        self
+    }
+
+    pub fn count(mut self, count: NonZeroU32) -> Self {
+        self.count = Some(count);
+        self
+    }
+
+    pub fn build(
+        &self,
+        binding: u32,
+        default_visibility: ShaderStages,
+    ) -> wgpu::BindGroupLayoutEntry {
+        wgpu::BindGroupLayoutEntry {
+            binding,
+            ty: self.ty,
+            visibility: self.visibility.unwrap_or(default_visibility),
+            count: self.count,
+        }
+    }
+}
+
+pub struct BindGroupLayoutEntries<const N: usize> {
+    entries: [wgpu::BindGroupLayoutEntry; N],
+}
+
+impl<const N: usize> BindGroupLayoutEntries<N> {
+    #[inline]
+    #[allow(unused)]
```

Comment:
why all the allow(unused)s?

Reply:
Oh, right, I started prorotyping it in a separate repo and clippy was complaining about unused. I'll remove it, it's not necessary.